dot/sync/message.go

```diff
@@ -92,51 +71,46 @@ func (s *Service) handleAscendingRequest(req *network.BlockRequestMessage) (*net
 		return nil, ErrInvalidBlockRequest
 	}
 
-	if endHash == nil {
-		endNumber = startNumber + max - 1
-		bestBlockNumber, err := s.blockState.BestBlockNumber()
-		if err != nil {
-			return nil, fmt.Errorf("failed to get best block %d for request: %w", bestBlockNumber, err)
-		}
-
-		if endNumber > bestBlockNumber {
-			endNumber = bestBlockNumber
-		}
-	} else {
-		header, err := s.blockState.GetHeader(*endHash)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get end block %s: %w", *endHash, err)
-		}
+	endNumber = startNumber + max - 1
+	bestBlockNumber, err := s.blockState.BestBlockNumber()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get best block %d for request: %w", bestBlockNumber, err)
```

Comment:
nit change wrapping prefix + do not add `bestBlockNumber` to context which is likely `0` if the call errors out

```suggestion
		return nil, fmt.Errorf("getting best block number for request: %w", err)
```